modules: mbedtls: remove default-enabling of hash algorithms

tomi-font opened the pull request:
Do not enable hash algorithms except SHA-256 by default.
This unnecessarily inflates the final code size even if not all the enabled hash algorithms are actually used.
    
SHA-256 is (for now) kept enabled by default because many configurations across the code base assume that there is some hash algorithm available without needing to enable it.

Relates to (and will help) https://github.com/zephyrproject-rtos/zephyr/pull/71947.

tomi-font committed:
modules: mbedtls: remove default-enabling of hash algorithms

Do not enable hash algorithms except SHA-256 by default.
This unnecessarily inflates the final code size even if not all the
enabled hash algorithms are actually used.

SHA-256 is (for now) kept enabled by default because many configurations
across the code base assume that there is some hash algorithm
available without needing to enable it.

Signed-off-by: Tomi Fontanilles <tomi.fontanilles@nordicsemi.no>

ceolin commented:
There are some tests failing in CI due some tangled build options, but I think that is the right direction. Features should be disabled by default and explicitly requested.

tomi-font committed:
drivers: crypto: mbedtls: fix dependency

Explicitly enable SHA-512 now that it is not enabled by default
anymore.

Signed-off-by: Tomi Fontanilles <tomi.fontanilles@nordicsemi.no>

tomi-font committed:
samples: tfm_integration: psa_crypto: improve the MbedTLS configuration

Move supported configuration items from the custom MbedTLS user
configuration file over to the Kconfig overlay.
The correct macros are then defined by the MbedTLS module according
to the values of the Kconfig options.

Signed-off-by: Tomi Fontanilles <tomi.fontanilles@nordicsemi.no>

ithinuel commented on west.yml:
```diff
@@ -277,12 +279,13 @@ manifest:
         - crypto
     - name: mcuboot
-      revision: 78bfe750cde0f8dfcf81bb4c0a68243d45906def
+      remote: mcu-tools
+      revision: d4394c2f9b76e0a7b758441cea3a8ceb896f66c8
```
Is my understanding correct that this implies that it is no longer relying on this mirror: https://github.com/zephyrproject-rtos/mcuboot/ ?

I don't see this change advertised in the commit message. Would it be worth adding a more visible note about this change?

tomi-font replied:
This is a temporary change to be able to pull the changes I made to upstream MCUboot. Zephyr's fork of it hasn't been updated yet, so the upstream remote needs to be used. (The change from `pull/[...]` to the hash is because my PR got merged upstream).
This will be updated once Zephyr's fork of MCUboot is updated, which will happen before this PR can be merged.

ceolin commented:
Changes on Zephyr looks fine for me. Lets get modules in and update it.

tomi-font commented:
This is ready for review.

fabiobaltieri commented:
@tomi-font looks like some TLS tests started failing after this got merged https://github.com/zephyrproject-rtos/zephyr/runs/24575825044, could you look into it?

tomi-font commented:
@fabiobaltieri Thanks for the heads-up. I opened a fix PR: #72334

aescolar commented:
@tomi-font would this PR warrant a line in the migration guidelines?

tomi-font commented:
> @tomi-font would this PR warrant a line in the migration guidelines?

Yeah, actually I started thinking about this afterwards. It does seem like a good idea. There will be other related PRs and changes made to the configuration; I'll address this point in a follow-up PR.
